Snapshot test fails with distributed runs

**Describe the bug**
Our nightly testing is failing in distributed settings with the snapshot test. All the nightly testing runs at 16 locales, but I suspect it could be reproduced with a smaller number of locales

**To Reproduce**
Steps to reproduce the behavior:
1. Build in multilocale
2. Run the test snapshot test with 16 locales (maybe less, not sure)

Probably will get most useful information with server in one terminal and then run the test in another with `ARKOUDA_RUNNING_MODE=CLIENT python3 -m pytest tests/io_test.py -k "test_snapshot"`


**Error Message**
```
_____________________________ IOTest.test_snapshot _____________________________

self = <io_test.IOTest testMethod=test_snapshot>

    def test_snapshot(self):
        from pandas.testing import assert_frame_equal
        df = ak.DataFrame(
            {
                "int_col": ak.arange(10),
                "uint_col": ak.array([i + 2**63 for i in range(10)], dtype=ak.uint64),
                "float_col": ak.linspace(-3.5, 3.5, 10),
                "bool_col": ak.randint(0, 2, 10, dtype=ak.bool),
                "bigint_col": ak.array([i + 2**200 for i in range(10)], dtype=ak.bigint),
                "segarr_col": ak.SegArray(ak.arange(0, 20, 2), ak.randint(0, 3, 20)),
                "str_col": ak.random_strings_uniform(0, 3, 10),
            }
        )
        df_str_idx = df.copy()
        df_str_idx._set_index(["A" + str(i) for i in range(len(df))])
        col_order = df.columns
        df_ref = df.to_pandas()
        df_str_idx_ref = df_str_idx.to_pandas()
        a = ak.randint(0, 10, 100)
        a_ref = a.to_list()
        s = ak.random_strings_uniform(0, 5, 50)
        s_ref = s.to_list()
        c = ak.Categorical(s)
        c_ref = c.to_list()
        g = ak.GroupBy(a)
        g_ref = {
            "perm": g.permutation.to_list(),
            "keys": g.keys.to_list(),
            "segments": g.segments.to_list(),
        }

        with tempfile.TemporaryDirectory(dir=IOTest.io_test_dir) as tmp_dirname:
            ak.snapshot(f"{tmp_dirname}/arkouda_snapshot_test")
            # delete variables
            del df
            del df_str_idx
            del a
            del s
            del c
            del g

            # verify no longer in the namespace
            with self.assertRaises(NameError):
                self.assertTrue(not df)
            with self.assertRaises(NameError):
                self.assertTrue(not df_str_idx)
            with self.assertRaises(NameError):
                self.assertTrue(not a)
            with self.assertRaises(NameError):
                self.assertTrue(not s)
            with self.assertRaises(NameError):
                self.assertTrue(not c)
            with self.assertRaises(NameError):
                self.assertTrue(not g)

            # restore the variables
            data = ak.restore(f"{tmp_dirname}/arkouda_snapshot_test")
            for vn in ["df", "df_str_idx", "a", "s", "c", "g"]:
                # ensure all variable names returned
                self.assertTrue(vn in data.keys())

            # validate that restored variables are correct
            self.assertTrue(
>               assert_frame_equal(df_ref[col_order], data["df"].to_pandas()[col_order]) is None
            )

tests/io_test.py:1283:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas/_libs/testing.pyx:52: in pandas._libs.testing.assert_almost_equal
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   AssertionError: DataFrame.iloc[:, 0] (column name="int_col") are different
E
E   DataFrame.iloc[:, 0] (column name="int_col") values are different (70.0 %)
E   [index]: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
E   [left]:  [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
E   [right]: [4, 1, 6, 3, 0, 7, 8, 2, 5, 9]
E   At positional index 0, first diff: 0 != 4

pandas/_libs/testing.pyx:172: AssertionError
```

--- a/arkouda/dataframe.py
+++ b/arkouda/dataframe.py
@@ -2539,7 +2539,7 @@ class DataFrame(UserDict):
         columns = {}
         for k, create_data in data.items():
             if k == "index":
-                idx = Index(create_pdarray(data["permutation"]))
+                idx = Index(create_pdarray(create_data))
             else:
                 comps = create_data.split("+|+")
                 if comps[0] == pdarray.objType.upper():

--- a/arkouda/dataframe.py
+++ b/arkouda/dataframe.py
@@ -347,7 +347,6 @@ class DataFrame(UserDict):
                 self._set_index(arange(self._size))
             else:
                 self._set_index(index)
-
             self.update_size()
 
     def __getattr__(self, key):
@@ -2538,10 +2537,13 @@ class DataFrame(UserDict):
         idx = None
         columns = {}
         for k, create_data in data.items():
-            if k == "index":
-                idx = Index(create_pdarray(create_data))
+            comps = create_data.split("+|+")
+            if k.lower() == "index":
+                if comps[0] == Strings.objType.upper():
+                    idx = Index(Strings.from_return_msg(comps[1]))
+                else:
+                    idx = Index(create_pdarray(comps[1]))
             else:
-                comps = create_data.split("+|+")
                 if comps[0] == pdarray.objType.upper():
                     columns[k] = create_pdarray(comps[1])
                 elif comps[0] == Strings.objType.upper():

--- a/arkouda/io.py
+++ b/arkouda/io.py
@@ -1984,4 +1984,4 @@ def restore(filename):
     objects in HDF5. Thus, they are returned within the dictionary as a dataframe.
     """
     restore_files = glob.glob(f"{filename}_SNAPSHOT_LOCALE*")
-    return read_hdf(restore_files)
+    return read_hdf(sorted(restore_files))

--- a/tests/io_test.py
+++ b/tests/io_test.py
@@ -1234,7 +1234,7 @@ class IOTest(ArkoudaTest):
         df_str_idx._set_index(["A" + str(i) for i in range(len(df))])
         col_order = df.columns
         df_ref = df.to_pandas()
-        df_str_idx_ref = df_str_idx.to_pandas()
+        df_str_idx_ref = df_str_idx.to_pandas(retain_index=True)
         a = ak.randint(0, 10, 100)
         a_ref = a.to_list()
         s = ak.random_strings_uniform(0, 5, 50)
@@ -1280,10 +1280,10 @@ class IOTest(ArkoudaTest):
 
             # validate that restored variables are correct
             self.assertTrue(
-                assert_frame_equal(df_ref[col_order], data["df"].to_pandas()[col_order]) is None
+                assert_frame_equal(df_ref[col_order], data["df"].to_pandas(retain_index=True)[col_order]) is None
             )
             self.assertTrue(
-                assert_frame_equal(df_str_idx_ref[col_order], data["df_str_idx"].to_pandas()[col_order]) is None
+                assert_frame_equal(df_str_idx_ref[col_order], data["df_str_idx"].to_pandas(retain_index=True)[col_order]) is None
             )
             self.assertListEqual(a_ref, data["a"].to_list())
             self.assertListEqual(s_ref, data["s"].to_list())

